packages/runner/cypress/integration/issues/issue-8350.js

```diff
@@ -0,0 +1,20 @@
+const { createCypress } = require('../../support/helpers')
+const { runIsolatedCypress } = createCypress()
+
+// https://github.com/cypress-io/cypress/issues/8350
+describe('issue-8350', () => {
+  it('does not hang on nested hook', () => {
```

Comment:
Should we add more tests for different cases like `beforeEach` in `beforeEach`, `afterEach` in `beforeEach`, `before` in `before`?

Reply:
the bug occurs when a hook is registered after the test has started, so it applies to more than just nested hooks. for that reason I think this covers it